server/src/domain/media/media.service.ts

```diff
@@ -180,8 +180,8 @@ export class MediaService {
 
     const allTargetsMatching = isTargetVideoCodec && isTargetAudioCodec && isTargetContainer;
 
-    const targetResolution = Number.parseInt(ffmpegConfig.targetResolution);
-    const isLargerThanTargetResolution = Math.min(videoStream.height, videoStream.width) > targetResolution;
+    const targetRes = Number.parseInt(ffmpegConfig.targetResolution) || Math.min(videoStream.height, videoStream.width);
+    const isLargerThanTargetRes = Math.min(videoStream.height, videoStream.width) > targetRes;
```

Comment:
This is not very clear to me. You are relying on parseInt to fail to parse `original` for this code to work properly. Can we add a more clear/obvious check for the text `original` instead? Specifically, leave the existing code as is and add a new variable is original then set isLargerThanTargetRes to the original condition  and not isOriginal. Maybe worth extracting get target resolution into a separate method for clarity too.

Reply:
The issue I saw was that `targetResolution` isn't actually a valid target resolution if `original` is set. I guess it's fine though since it's a dropdown and not user input.

Reply:
Validating user input for that field is more of an unrelated issue IMO. In this PR we're adding a known value of "original", which I think should have specific behavior linked to it, which should be easy to follow in the code here. I think unrelated we can add some logic for what to do when the `targetResolution` is invalid, which _could_ also be to not adjust it. Although, the better behavior might be to fallback to the default value (`'720'`) instead.